src/plugins/mpi/pmix/pmixp_agent.c

```diff
@@ -155,13 +156,10 @@ static int _abort_conn_read(eio_obj_t *obj, List objs)
 {
 	struct sockaddr_in abort_client;
 	int abort_client_sock;
-	uint32_t abort_client_len = sizeof(abort_client);
-	uint32_t ret_status;
-
-	int fd;
+	char ret_status[32];
```

Comment:
I'll suggest to keep `ret_status` as integer type, but using it as `(char *)ret_status` to avoid compilation warnings:
```C
if ((len = slurm_read_stream(abort_client_sock, (char *)&ret_status, sizeof(ret_status))) == -1)
```

Reply:
Yes, this is a very bad idea to have it as char.
@BorisBochkaryov I thought we already went through this and agreed that it will be uint32_t.
Where this char thing came from again??
Also, why is it char[32]??? uint32_t is 4B only!